src/main/java/hudson/plugins/promoted_builds/parameters/PromotedBuildParameterDefinition.java

```diff
@@ -127,12 +175,10 @@ public List getBuilds() {
         }
 
         for (Iterator iter = job.getBuilds().iterator(); iter.hasNext(); )  {
-            Run run = (Run) iter.next();
-
-            List buildActions = run.getActions(PromotedBuildAction.class);
-            for (int i = 0; i < buildActions.size(); i++) {
-                PromotedBuildAction buildAction = (PromotedBuildAction) buildActions.get(i);
+            final Run run = (Run) iter.next();
```

Comment:
Yes, we have to cast. There's a raw types issue, which would need a serious refactoring